sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -161,6 +161,78 @@ class CollationStringExpressionsSuite
     })
   }
 
+  test("Ascii & UnBase64 string expressions with collation") {
+    case class AsciiUnBase64TestCase[R](q: String, dt: DataType, r: R)
+    val testCases = Seq(
+      AsciiUnBase64TestCase("select ascii('a' collate utf8_binary)", IntegerType, Row(97)),
+      AsciiUnBase64TestCase("select ascii('a' collate utf8_binary_lcase)", IntegerType, Row(97)),
+      AsciiUnBase64TestCase("select unbase64('YQ==' collate utf8_binary)", BinaryType,
+        Row(Seq(97))),
+      AsciiUnBase64TestCase("select unbase64('YQ==' collate utf8_binary_lcase)", BinaryType,
+        Row(Seq(97)))
+    )
+    testCases.foreach(t => {
+      // Result & data type
+      checkAnswer(sql(t.q), t.r)
```

Comment:
```suggestion
      checkAnswer(sql(t.q), Row(t.r))
```

(and other places too, to keep tests consistent)